apis/v1alpha1/patch_baseline.go

```diff
@@ -0,0 +1,132 @@
+	// that should be applied to the managed nodes. The default value is false.
+	// Applies to Linux managed nodes only.
+	ApprovedPatchesEnableNonSecurity *bool `json:"approvedPatchesEnableNonSecurity,omitempty"`
+	// User-provided idempotency token.
+	ClientToken *string `json:"clientToken,omitempty"`
+	// A description of the patch baseline.
+	Description *string `json:"description,omitempty"`
+	// A set of global filters used to include patches in the baseline.
+	GlobalFilters *PatchFilterGroup `json:"globalFilters,omitempty"`
+	// The name of the patch baseline.
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
+	// Defines the operating system the patch baseline applies to. The default value
+	// is WINDOWS.
+	OperatingSystem *string `json:"operatingSystem,omitempty"`
+	// A list of explicitly rejected patches for the baseline.
+	//
+	// For information about accepted formats for lists of approved patches and
+	// rejected patches, see About package name formats for approved and rejected
+	// patch lists (https://docs.aws.amazon.com/systems-manager/latest/userguide/patch-manager-approved-rejected-package-name-formats.html)
+	// in the Amazon Web Services Systems Manager User Guide.
+	RejectedPatches []*string `json:"rejectedPatches,omitempty"`
+	// The action for Patch Manager to take on patches included in the RejectedPackages
+	// list.
+	//
+	//   - ALLOW_AS_DEPENDENCY : A package in the Rejected patches list is installed
+	//     only if it is a dependency of another package. It is considered compliant
+	//     with the patch baseline, and its status is reported as InstalledOther.
+	//     This is the default action if no option is specified.
+	//
+	//   - BLOCK : Packages in the RejectedPatches list, and packages that include
+	//     them as dependencies, aren't installed under any circumstances. If a package
+	//     was installed before it was added to the Rejected patches list, it is
+	//     considered non-compliant with the patch baseline, and its status is reported
+	//     as InstalledRejected.
+	RejectedPatchesAction *string `json:"rejectedPatchesAction,omitempty"`
+	// Information about the patches to use to update the managed nodes, including
+	// target operating systems and source repositories. Applies to Linux managed
+	// nodes only.
+	Sources []*PatchSource `json:"sources,omitempty"`
+	// Optional metadata that you assign to a resource. Tags enable you to categorize
+	// a resource in different ways, such as by purpose, owner, or environment.
+	// For example, you might want to tag a patch baseline to identify the severity
+	// level of patches it specifies and the operating system family it applies
+	// to. In this case, you could specify the following key-value pairs:
+	//
+	//   - Key=PatchSeverity,Value=Critical
+	//
+	//   - Key=OS,Value=Windows
+	//
+	// To add tags to an existing patch baseline, use the AddTagsToResource operation.
+	Tags []*Tag `json:"tags,omitempty"`
```

Comment:
Looks like the update API doesn't update tags. Can we support  tag updates for this fields?

Reply:
Done